src/core/vector.pmt

```diff
@@ -898,7 +898,86 @@ long int FUNCTION(igraph_vector, qsort_ind)(TYPE(igraph_vector) *v,
     return 0;
 }
 
-#endif
+/**
+ * \function igraph_vector_lex_cmp
+ * \brief Lexicographical comparison of two vectors.
+ *
+ * \param lhs The first vector.
+ * \param rhs The second vector.
+ * \return -1 if \p lhs is lexicographically smaller,
+ *         0 if \p lhs and \p rhs are equal, else 1.
+ *
+ * Time complexity: O(n), the number of elements in the smaller vector.
+ */
+int FUNCTION(igraph_vector, lex_cmp)(const void *lhs, const void *rhs) {
+    const TYPE(igraph_vector) *a = lhs;
+    const TYPE(igraph_vector) *b = rhs;
+    long int i, sa, sb;
+
+    sa = FUNCTION(igraph_vector, size)(a);
+    sb = FUNCTION(igraph_vector, size)(b);
+
+    for (i = 0; i < sa; i++) {
+        if (i >= sb) {
+            /* b is shorter, and equal to the first part of a */
+            return 1;
+        }
+        if (VECTOR(*a)[i] < VECTOR(*b)[i]) {
+            return -1;
+        }
+        if (VECTOR(*a)[i] > VECTOR(*b)[i]) {
+            return 1;
+        }
+    }
+    if (i == sb) {
+        return 0;
+    }
+    /* a is shorter, and equal to the first part of b */
+    return -1;
+}
+
+/**
+ * \function igraph_vector_lex_cmp_rev
+ * \brief Reverse lexicographical comparison of two vectors.
+ *
```

Comment:
Let us spell out that this compares vectors starting with the last element. Someone may misinterpret "reverse" as sorting in reverse order.